### DISCUSSION TITLE
phoneme embeddings

### DISCUSSION THREAD

**Discussion by nellorebhanuteja** [espnet/espnet#4350] [Q&A] [answered]
Where is the code that extracts phoneme embeddings in espnet2 TTS?

**Comment by nellorebhanuteja** [comment, marked as answer]
@kan-bayashi

**Reply by kan-bayashi** [reply]
You mean ids -> embedding? or text -> ids?

**Reply by nellorebhanuteja** [reply]
the embeddings for text we use in Fastspeech 2

**Reply by kan-bayashi** [reply]
https://github.com/espnet/espnet/blob/beb3360276aa9ff65fe84f4c5e99c0c063c2a6be/espnet2/tts/fastspeech2/fastspeech2.py#L489-L560
https://github.com/espnet/espnet/blob/beb3360276aa9ff65fe84f4c5e99c0c063c2a6be/espnet2/tts/fastspeech2/fastspeech2.py#L631
https://github.com/espnet/espnet/blob/beb3360276aa9ff65fe84f4c5e99c0c063c2a6be/espnet/nets/pytorch_backend/conformer/encoder.py#L255

**Reply by nellorebhanuteja** [reply]
Thank you!
